### DISCUSSION TITLE
Smartthings integration

### DISCUSSION THREAD

**Discussion by rogueturnip** [taligentx/dscKeybusInterface#19] [dscKeybusInterface Discussion]
I don't have time right now to do this integration but if someone else has interest these guys released a bunch of device handlers for their implementation and it should be relatively simple to modify them to work with this design.
https://github.com/konnected-io/konnected-security

**Comment by taligentx** [comment]
Thanks, SmartThings integration would be useful.  I looked into this a few weeks ago and wasn't able to find a way to report when the alarm was triggered - from their forums, some of the third party SmartThings devs state that this part of the API is closed off except for paid partners like Scout/ADT, etc:

https://community.smartthings.com/t/getting-the-shm-status-through-the-api/41285/7

Do you know if Konnected is able to send triggered alarm states to Smart Home Monitor?  Otherwise it should be possible to get basic armed/disarmed states to SHM - the question will be if there's a useful purpose without notifications on the alarm being triggered.

**Comment by rogueturnip** [comment]
I believe you are correct in your assessment that you can't trigger the alarm, however the solution for this (we did this for a different integration) is to create a device handler called "alarm".  Think of this as your virtual siren.  With this device handler you just send the state changes to that device handler and then you can create rules based on the state.. 

One of the big useful purposes of this integration would be to pass the zone states over to smartthings so that those can be leveraged in the smart home as sensors to allow for other automation.  AN integration like this would immediately give most homes 8-20 sensors.

**Comment by taligentx** [comment]
Another possible method:

https://github.com/DanielOgorchock/ST_Anything
